src/Stratis.Bitcoin/P2P/SelfEndpointTracker.cs

```diff
@@ -11,6 +19,15 @@ public class SelfEndpointTracker : ISelfEndpointTracker
         /// <summary>Hashset to hold the endpoints currently known to be itself.</summary>
         private readonly ConcurrentHashSet<IPEndPoint> knownSelfEndpoints = new ConcurrentHashSet<IPEndPoint>();
 
+        /// <summary>External IP address of the node.</summary>
```

Comment:
Let's also introduced a logger – so have a constructor that accepts logger factory. It also move all the initialisation to constructor